### PR TITLE
Versioning, Packaging, Logging

### PR DESCRIPTION
**Pull request by KPrasch**
Prepares nucypher for alpha release and publication to PyPI

* `setup.py` script
* Package metadata, Trove Classification, Authorship, etc.
* Single source versioning
* bumpversion configuration
* Sentry Log collection
* Demo Polishing

### CODE REVIEW

File: nucypher/__about__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+__summary__ = 'A proxy re-encryption network to empower privacy in decentralized systems.',
+
+__version__ = "0.1.0-alpha.0"
+
+__author__ = "NuCypher"
+
+__email__ = "dev@nucypher.com"
+
+__license__ = "GNU General Public License, Version 3"
```
Comment (tuxxy) [inline nucypher/__about__.py]: Looks good.

File: .bumpversion.cfg [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	alpha
+	beta
+	rc
+	stable
+
+[bumpversion:part:devnum]
+
+[bumpversion:file:README.md]
+
+[bumpversion:file:umbral/__about__.py]
```
Comment (cygnusv) [inline .bumpversion.cfg]:
```suggestion
[bumpversion:file:nucypher/__about__.py]
```

File: setup.py [hunk 3]
```diff
@@ -1,19 +1,134 @@
...
+    'bytestringSplitter',
+    'hendrix>=3.1.0',
+
+    # Third Party (General)
+    'cryptography>=2.3',
+    'pysha3',
+    'requests',
+    'sqlalchemy',
+    'apistar==0.5.42',
+    'tzlocal==2.0.0b1',
```
Comment (michwill) [inline setup.py]: This doesn't work for me. As soon as `tzlocal==2.0.0b1` got into `setup.py`, `nucypher` command fails for me. It shouldn't be pinned to this version neither in `setup.py`, nor in `Pipfile`

Comment (tuxxy) [reply setup.py]: @KPrasch and I discussed this, taking care of this now is going to needlessly break some dev environments. We're going to push this for a later PR.

File: cli/main.py [hunk 4]
```diff
@@ -1,22 +1,22 @@
 #!/usr/bin/env python3
```
Comment (michwill) [inline cli/main.py]: Wait... I guess, cli should be a module (so it should have `__init__.py`). Also, it should probably live in nucypher namespace.

Currently, I am installing in virtualenv (so no Pipenv to help us, like in real world), and:
```
  File "/home/michwill/Projects/nucypher-kms/_test/bin/nucypher", line 7, in <module>
    from cli.main import cli
ModuleNotFoundError: No module named 'cli'
```

File: setup.py (hunk 3 above)
Comment (michwill) [reply setup.py]: Sure. I guess, we should have an issue for this, or add a comment in an open PR where this is to be done, in order to not forget.
Although, it wouldn't even install for me without me fixing it every time, so I won't forget :joy:

File: cli/main.py (hunk 4 above)
Comment (jMyles) [reply cli/main.py]: Yeah, I think so too.  But not today.

**Pull request merged**